test/helpers/framework.go

```diff
@@ -115,6 +114,64 @@ func (tc *E2ETestFramework) DeployLogGenerator() error {
 	return tc.waitForDeployment(namespace, "log-generator", defaultRetryInterval, defaultTimeout)
 }
 
+func (tc *E2ETestFramework) DeployJsonLogGenerator(vals map[string]string) (string, string, error) {
+	namespace := tc.CreateTestNamespace()
+	pycode := `
```

Comment:
nit. it would be nice to have this python code on a separate file, e.g. `test/files/jsonlogger.py` to enable better editing on editors for python.

Reply:
i think it is easier to see the generator code in-place rather than in a different file. the function accepts arguments and uses them to generate the logic. in a different file, there is a disconnect.